optuna/storages/_rdb/storage.py

```diff
@@ -1098,9 +1098,11 @@ def _init_alembic(self) -> None:
 
     def _set_alembic_revision(self, revision: str) -> None:
 
-        context = alembic_migration.MigrationContext.configure(self.engine.connect())
-        script = self._create_alembic_script()
-        context.stamp(script, revision)
+        connection = self.engine.connect()
+        context = alembic_migration.MigrationContext.configure(connection)
+        with connection.begin():
+            script = self._create_alembic_script()
+            context.stamp(script, revision)
```

Comment:
```
tests/storages_tests/rdb_tests/test_storage.py: 11 warnings
  /Users/c-bata/go/src/github.com/optuna/optuna/optuna/storages/_rdb/storage.py:1103: RemovedIn20Warning: The current statement is being autocommitted using implicit autocommit, which will be removed in SQLAlchemy 2.0. Use the .begin() method of Engine or Connection in order to use an explicit transaction for DML and DDL statements. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
    context.stamp(script, revision)
```

Reply:
[Note] I confirmed that this logic worked with the following test case locally.
https://github.com/optuna/optuna/blob/4d06898dce1fa0ddbc07eb080d55f69a63362fe5/tests/storages_tests/rdb_tests/test_storage.py#L139-L142.

Maybe we can activate it in a follow-up PR.

Reply:
Good catch!